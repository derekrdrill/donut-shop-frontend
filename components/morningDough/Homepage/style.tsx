import styled from 'styled-components';
import { Grid } from '@mui/material';

export const HomepageRoot = styled(Grid)({
  overflowX: 'hidden',
});

export const HomepageImage = styled.img<{
  $isOnRightSide?: boolean;
  $isOnLeftSide?: boolean;
}>(({ $isOnLeftSide, $isOnRightSide }) => ({
  height: 300,
  width: 300,
  transform: $isOnLeftSide ? 'translateX(-100px)' : $isOnRightSide ? 'translate(100px)' : 'none',
}));

export const HomepageText = styled.h1<{
  $fontColor?: string;
  $fontSize: number;
  $isSubtext?: boolean;
}>(({ $fontColor, $fontSize, $isSubtext }) => ({
  color: $fontColor,
  fontFamily: $isSubtext ? `'Josefin Sans', sans-serif` : `'Lilita One', cursive !important`,
  fontSize: $fontSize,
  margin: '15px 0',
  textAlign: 'center',
  textTransform: 'uppercase',
}));

export const HomePageTextSpan = styled.span<{ $fontColor: string }>(({ $fontColor }) => ({
  color: $fontColor,
}));
