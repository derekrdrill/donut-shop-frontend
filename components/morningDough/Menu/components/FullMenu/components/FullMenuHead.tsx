import * as React from 'react';
import { Grid } from '@mui/material';

import {
  FullMenuHeadContainer,
  FullMenuHeadImage,
  FullMenuHeadTextItem,
  FullMenuText,
} from '../style';

import icedCoffee from '../../../assets/images/coffeeAndTea/icedCoffee.png';
import chocolateFrostingSprinklesDonut from '../../../assets/images/donuts/chocolateFrostingSprinklesDonut.png';
import everythingBagel from '../../../assets/images/bread/everythingBagel.png';
import hotLatte from '../../../assets/images/coffeeAndTea/hotLatte.png';

const FullMenuHead = () => (
  <Grid container>
    <Grid item xs={12}>
      <FullMenuHeadContainer container>
        <Grid item xs={12} sm={6} md={3} order={{ xs: 1, lg: 0 }}>
          <Grid container justifyContent={{ xs: 'center', lg: 'flex-start' }}>
            <Grid item xs={6} md={12} lg={6}>
              <FullMenuHeadImage src={icedCoffee} />
            </Grid>
            <Grid item xs={6} md={12} lg={6} display={{ xs: 'block', lg: 'none' }}>
              <FullMenuHeadImage src={everythingBagel} />
            </Grid>
          </Grid>
        </Grid>
        <FullMenuHeadTextItem item xs={12} md={6} order={{ xs: 0, lg: 1 }}>
          <FullMenuText textAlign={{ xs: 'left', lg: 'center' }} variant='h2' $fontColor='#f2755f'>
            FIRE UP THAT ORDER ðŸ”¥
          </FullMenuText>
          <FullMenuText textAlign={{ xs: 'left', lg: 'center' }} variant='h3' $fontColor='#6a017f'>
            We'll be ready ðŸ˜Ž
          </FullMenuText>
          <FullMenuText
            textAlign={{ xs: 'left', lg: 'center' }}
            variant='h6'
            $fontColor='#1c1c1c'
            $isSubtext
          >
            Special tasty treats just for you
          </FullMenuText>
          <FullMenuText
            textAlign={{ xs: 'left', lg: 'center' }}
            variant='body2'
            $fontColor='#5e5e5e'
            $isSubtext
          >
            $1 croissants or FREE small coffee with the pruchase of an avacado toast. Price and
            participation may vary. Limited time offer. Terms apply.
          </FullMenuText>
        </FullMenuHeadTextItem>
        <Grid item xs={12} sm={6} md={3} order={2}>
          <Grid container justifyContent={{ lg: 'center' }}>
            <Grid item xs={6} md={12} lg={6}>
              <FullMenuHeadImage src={chocolateFrostingSprinklesDonut} />
            </Grid>
            <Grid item xs={6} md={12} lg={6} display={{ xs: 'block', lg: 'none' }}>
              <FullMenuHeadImage src={hotLatte} />
            </Grid>
          </Grid>
        </Grid>
      </FullMenuHeadContainer>
    </Grid>
  </Grid>
);

export default FullMenuHead;
