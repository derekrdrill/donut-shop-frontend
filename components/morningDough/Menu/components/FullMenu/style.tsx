import { Grid, Typography } from '@mui/material';
import styled from 'styled-components';

export const FullMenuHeadContainer = styled(Grid)({
  backgroundColor: '#fef3f1',
  minHeight: 250,
  opacity: 0.8,
});

export const FullMenuHeadTextItem = styled(Grid)({
  padding: 20,
});

export const FullMenuContainer = styled(Grid)({
  paddingBottom: 40,
  margin: '40px 4% 0px 4%',
});

export const FullMenuText = styled(Typography)<{
  $fontColor?: string;
  $isSubtext?: boolean;
}>(({ $fontColor, $isSubtext }) => [
  {
    color: $fontColor,
    fontFamily: `'Lilita One', cursive !important`,
    margin: '15px 0',
  },
  $isSubtext && {
    fontFamily: `'Rubik', sans-serif !important`,
  },
]);

export const FullMenuHeadImage = styled.img({
  height: 225,
  width: 225,
});

export const FullMenuItemRoot = styled(Grid)({
  height: 200,
  padding: 10,
});

export const FullMenuItemContainer = styled(Grid)({
  ':hover': {
    backgroundColor: '#f8d7ff',
  },
  border: '1px solid #ababab',
  borderRadius: 10,
  cursor: 'pointer',
  maxWidth: 340,
  transition: 'all 250ms ease-in',
});

export const FullMenuItemImage = styled.img({
  height: 150,
  width: 150,
});
