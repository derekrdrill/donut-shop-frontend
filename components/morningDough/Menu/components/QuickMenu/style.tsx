import styled from 'styled-components';
import { Grid } from '@mui/material';

export const QuickMenuImage = styled.img({
  height: 100,
  transition: 'all 50ms ease-in',
  width: 100,
});

export const QuickMenuTabContainer = styled(Grid)<{ $isMenuTabActive?: boolean }>(
  ({ $isMenuTabActive }) => ({
    '&:hover': {
      backgroundColor: '#bb00e0',
      color: '#FFFFFF',
      opacity: !$isMenuTabActive ? 0.6 : 1,
    },
    backgroundColor: $isMenuTabActive ? '#bb00e0' : 'inherit',
    color: $isMenuTabActive ? '#FFFFFF' : 'inherit',
    cursor: 'pointer',
    opacity: 0.8,
    padding: 8,
  }),
);

export const QuickMenuItemContainer = styled(Grid)({
  '&:hover': {
    img: {
      opacity: 0.8,
    },
    '.MuiTypography-root': {
      textDecoration: 'underline',
    },
    border: '1px gainsboro solid',
    borderRadius: 20,
    boxShadow: 'rgba(0, 0, 0, 0.16) 0px 3px 6px, rgba(0, 0, 0, 0.23) 0px 3px 6px',
  },
  cursor: 'pointer',
  maxHeight: 185,
  minWidth: 100,
  padding: 15,
});

export const QuickViewMenuItemsRoot = styled(Grid)({
  backgroundColor: '#f5f5f5',
  height: 354,
  overflowY: 'auto',
  padding: 20,
});
