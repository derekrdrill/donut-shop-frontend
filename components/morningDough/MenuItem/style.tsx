import styled from 'styled-components';
import { Button, Grid, ToggleButton, Typography } from '@mui/material';
import ActionButton from '../../common/ActionButton/ActionButton';

export const MenuItemImage = styled.img<{ height: number; width: number }>(({ height, width }) => ({
  height: height,
  widht: width,
}));

export const MenuItemDetailContainer = styled(Grid)({
  minHeight: 500,
  padding: '40px 60px',
});

export const MenuItemDetailText = styled(Typography)<{
  $fontColor?: string;
  $isSubtext?: boolean;
}>(({ $fontColor, $isSubtext }) => [
  {
    color: $fontColor,
    fontFamily: `'Lilita One', cursive !important`,
    margin: '15px 0',
  },
  $isSubtext && {
    fontFamily: `'Rubik', sans-serif !important`,
  },
]);

export const MenuItemToggleButton = styled(ToggleButton)({
  width: 75,
});

export const MenuItemOrderButton = styled(ActionButton)({
  borderRadius: '20px !important',
});
