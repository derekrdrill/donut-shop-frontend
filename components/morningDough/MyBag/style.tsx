import styled from 'styled-components';
import { Badge, Grid } from '@mui/material';
import ShoppingBagIcon from '@mui/icons-material/ShoppingBag';

export const MyBagRoot = styled(Grid)({
  marginTop: 40,
});

export const MyBagIconSVG = styled(ShoppingBagIcon)({
  marginTop: 7,
  transition: 'all 250ms linear',
  transform: 'scale(1.4)',
});

export const MyBagIconBadge = styled(Badge)({
  '.MuiBadge-badge': {
    backgroundColor: 'white',
    color: '#6a017f',
    border: '1px #6a017f solid',
    transition: 'all 250ms linear',
  },
});

export const MyBagCard = styled(Grid)({
  border: '1px solid gainsboro',
  borderRadius: 5,
  margin: '0px 10px !important',
  padding: 20,
});

export const MyBagItemContainer = styled(Grid)({
  height: 390,
  overflowY: 'auto',
});

export const MyBagItemRow = styled(Grid)({
  borderBottom: '1px solid gainsboro',
  borderTop: '1px solid gainsboro',
  padding: 15,
});

export const MyBagItemImage = styled.img({
  height: 40,
  width: 40,
});

export const MyBagIconContainer = styled(Grid)({
  '&:hover': {
    '.MuiSvgIcon-root': {
      color: '#a501c6',
      transform: 'scale(1.6)',
    },
    '.MuiBadge-root': {
      transform: 'scale(1.2)',
      zIndex: 4,
    },
  },
  cursor: 'pointer',
});

export const MyBagPaymentContainer = styled(Grid)({
  borderTop: '1px solid gainsboro',
  padding: 15,
});
