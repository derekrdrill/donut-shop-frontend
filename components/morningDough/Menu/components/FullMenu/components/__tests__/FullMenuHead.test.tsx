import '@testing-library/jest-dom';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import FullMenuHead from '../FullMenuHead';

import { FullMenuText } from '../../style';

describe('FullMenuHead unit tests', () => {
  it('renders as expected', () => {
    const result = renderer.create(<FullMenuHead />).toJSON();
    expect(result).toMatchSnapshot();
  });

  it('renders FullMenuText styled component', async () => {
    render(<FullMenuText data-testid='full-menu-text' />);

    const fullMenuText = await screen.findAllByTestId('full-menu-text');
    expect(fullMenuText[0]).toHaveClass('sc-hLseeU kfctxS');
  });

  it('renders FullMenuText styled component (isSubText)', async () => {
    render(<FullMenuText data-testid='full-menu-text' $isSubtext />);

    const fullMenuText = await screen.findAllByTestId('full-menu-text');
    expect(fullMenuText[0]).toHaveClass('sc-hLseeU hefLba');
  });
});
