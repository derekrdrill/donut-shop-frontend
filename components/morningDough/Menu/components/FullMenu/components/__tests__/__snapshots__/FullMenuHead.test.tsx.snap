// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullMenuHead unit tests renders as expected 1`] = `
<div
  className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container sc-beqWaB hTrTut css-11lq3yg-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3 css-1aljnjw-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-container css-68owf2-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-12 MuiGrid-grid-lg-6 css-1079s95-MuiGrid-root"
          >
            <img
              className="sc-eDDNvR gZzQUy"
              src={
                {
                  "process": [Function],
                }
              }
            />
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-12 MuiGrid-grid-lg-6 css-1xf4ifq-MuiGrid-root"
          >
            <img
              className="sc-eDDNvR gZzQUy"
              src={
                {
                  "process": [Function],
                }
              }
            />
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 sc-gueYoa jmZJof css-mb590u-MuiGrid-root"
      >
        <h2
          className="MuiTypography-root MuiTypography-h2 sc-hLseeU PzYec css-10sp2wv-MuiTypography-root"
        >
          FIRE UP THAT ORDER ðŸ”¥
        </h2>
        <h3
          className="MuiTypography-root MuiTypography-h3 sc-hLseeU hJUVBK css-1pm15ke-MuiTypography-root"
        >
          We'll be ready ðŸ˜Ž
        </h3>
        <h6
          className="MuiTypography-root MuiTypography-h6 sc-hLseeU cHiVpT css-1u98osu-MuiTypography-root"
        >
          Special tasty treats just for you
        </h6>
        <p
          className="MuiTypography-root MuiTypography-body2 sc-hLseeU bpCwal css-odxzrf-MuiTypography-root"
        >
          $1 croissants or FREE small coffee with the pruchase of an avacado toast. Price and participation may vary. Limited time offer. Terms apply.
        </p>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3 css-ax2l3g-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-container css-sbmkh9-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-12 MuiGrid-grid-lg-6 css-1079s95-MuiGrid-root"
          >
            <img
              className="sc-eDDNvR gZzQUy"
              src={
                {
                  "process": [Function],
                }
              }
            />
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-12 MuiGrid-grid-lg-6 css-1xf4ifq-MuiGrid-root"
          >
            <img
              className="sc-eDDNvR gZzQUy"
              src={
                {
                  "process": [Function],
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
